### ISSUE TITLE
`@typescript-eslint/*` errors on spreading flat configs

### ISSUE THREAD

**Issue by wKich-octra** [storybookjs/eslint-plugin-storybook#205]
**Describe the bug**
It seems type definition of storybook.configs['flat/*'] contains `any` in nested fields which causes lint errors.

**To Reproduce**
1. Define following eslint config:
  ```js
  import tseslint from 'typescript-eslint';
  import storybook from 'eslint-plugin-storybook';

  const config = tseslint.config(
    ...storybook.configs['flat/recommended'],
    ...tseslint.configs.strictTypeChecked,
    ...tseslint.configs.stylisticTypeChecked,
  );
  
  export default config;
  ```
2. Run eslint to lint its own config
4. See errors: `@typescript-eslint/no-unsafe-argument` and `@typescript-eslint/no-unsafe-member-access` on storybook's config spread line

**Expected behavior**
There should be no errors.

**Screenshots**
<img width="963" alt="Image" src="https://github.com/user-attachments/assets/99e4ed81-0de6-4dab-8ebd-74a4b4cda4c2" />

**Additional context**
Used versions:
- eslint-plugin-storybook - 9.0.0
- typescript-eslint - 8.33
- typescript - 5.8.3
- eslint - 9.27.0
